drivers/stepper/allegro/a4988.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Marvin Ouma
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT allegro_a4988
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/stepper.h>
+#include "../step_dir/step_dir_stepper_common.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(a4988, CONFIG_STEPPER_LOG_LEVEL);
+
+#define MSX_PIN_COUNT       3
+#define MSX_PIN_STATE_COUNT 5
+
+static enum stepper_micro_step_resolution a4988_msx_resolutions[MSX_PIN_STATE_COUNT] = {
+	STEPPER_MICRO_STEP_1, STEPPER_MICRO_STEP_2,  STEPPER_MICRO_STEP_4,
+	STEPPER_MICRO_STEP_8, STEPPER_MICRO_STEP_16,
+};
+
+struct a4988_config {
+	struct step_dir_stepper_common_config common;
+	struct gpio_dt_spec sleep_pin;
+	struct gpio_dt_spec enable_pin;
+	struct gpio_dt_spec reset;
+	struct gpio_dt_spec *msx_pins;
+	enum stepper_micro_step_resolution *msx_resolutions;
+};
+
+struct a4988_data {
+	struct step_dir_stepper_common_data common;
+	enum stepper_micro_step_resolution resolution;
+};
+
+STEP_DIR_STEPPER_STRUCT_CHECK(struct a4988_config, struct a4988_data);
+
+static int a4988_stepper_enable(const struct device *dev, const bool enable)
+{
+	/* enable and sleep pins need to be set logic low to be on  */
```

Comment:
Hey @Pancakem  the data sheet seems not to agree on this see page 10 on here https://www.pololu.com/file/download/A4988.pdf?file_id=0J450